src/coreclr/inc/corinfo.h

```diff
@@ -404,6 +404,7 @@ enum CorInfoHelpFunc
        which is the right helper to use to allocate an object of a given type. */
 
     CORINFO_HELP_NEWFAST,
+    CORINFO_HELP_NEWFAST_FROZEN,    // allocator for objects on a frozen segment
```

Comment:
This comment suggests that the returned object is guaranteed to be in a frozen segment. It is not the case. The returned object may be on the frozen segment, but there are no guarantees that it is actually going to be allocated there.

The comments should be updated at least. I am torn on whether it would be a good idea to change the name to `CORINFO_HELP_NEWFAST_MAYBEFROZEN` to make the contract more obvious.

Reply:
Addressed, MAYBEFROZEN sounds good to me, we already have helper calls with `_MAYBENULL` name